contentcuration/contentcuration/frontend/channelEdit/views/files/FileUpload.vue

```diff
@@ -164,16 +185,37 @@
         });
       },
       handleRemoveFile(file) {
-        this.deleteFile(file);
-        if (file.id === this.selected) {
-          this.selectFirstFile();
+        if (this.primaryFileCount > 1 && !this.isRemoveFileApproved) {
+          this.showRemoveFileWarning = true;
+        } else {
```

Comment:
This else branch confuses me a bit. If there's only one file, we delete the file without approval?

Reply:
That really seems backwards to me. If the user removes the last remaining file, then the node becomes incomplete. Therefore it would seem to make sense that it would confirm the deletion of a sole file. The link you shared is a reference to your new /changed code. The existing logic appears to allow removal when [there are more than one file](https://github.com/learningequality/studio/blob/8aea60aca21b02f4f8f0ed96b8b46205b1efef87/contentcuration/contentcuration/frontend/channelEdit/views/files/FileUpload.vue#L117-L119), so in that sense, it was most defensive when there was only one file, whereas your logic is defensive (requiring approval) when there are more than one and very lax with one file.

Reply:
Ahh I get your point! The current implementation doesn't allow removal of single files as the delete option isn't displayed in these instances.
![image](https://github.com/user-attachments/assets/08a30c91-abfd-45ce-8476-03f8f25d1c9e)

Maybe a more general question would be, do we want users deleting single files? If so, then it makes total sense to add another approval dialog (different from the approval for users that are removing extra files that we are moving away from) to warn users of the implications?

Reply:
Okay so it seems worthwhile to me to align this logic with that. In other words, have it check if file could be removed and if not, do nothing. As it's written, I feel it's confusing considering this behavior and aforementioned pathway